src/microbe_stage/Microbe.cs

```diff
@@ -1427,22 +1427,39 @@ private void OnIGotAddedToColony()
             return;
 
         var oldRotation = Rotation;
-        var vectorToParent = GlobalTransform.origin - ColonyParent.GlobalTransform.origin;
-        ChangeNodeParent(ColonyParent);
 
-        var vectorToParentRotated = vectorToParent.Rotated(Vector3.Down, Rotation.y);
-        var vectorToMembrane = Membrane.GetExternalOrganelle(vectorToParentRotated.x, vectorToParentRotated.y);
+        // Gets the global rotation of the parent
+        var globalParentRotation = ColonyParent.GlobalTransform.basis.GetEuler();
+
+        // A vector from the parent to me
+        var vectorFromParent = GlobalTransform.origin - ColonyParent.GlobalTransform.origin;
+
+        // This vector gets as if I had no rotation.
+        // This is important, because GetExternalOrganelle only works with non-rotated microbes
+        var vectorToParentRotated = (-vectorFromParent).Rotated(Vector3.Down, Rotation.y);
 
-        vectorToParentRotated = (-vectorToParent).Rotated(Vector3.Down, ColonyParent.Rotation.y);
-        var parentVectorToItsMembrane =
-            ColonyParent.Membrane.GetExternalOrganelle(vectorToParentRotated.x, vectorToParentRotated.y);
+        // This vector gets rotated as if the parent had no rotation.
+        var vectorFromParentRotated =
+            vectorFromParent.Rotated(Vector3.Down, globalParentRotation.y);
 
-        var requiredDistance = vectorToMembrane.Length() + parentVectorToItsMembrane.Length();
+        // Calculates the vector from my center to my membrane towards the parent.
+        // This vector gets rotated back, because I've rotated it two calls above.
+        var correctedVectorToParent = Membrane.GetExternalOrganelle(vectorToParentRotated.x, vectorToParentRotated.z)
+            .Rotated(Vector3.Up, Rotation.y);
 
-        var offset = vectorToParent.Normalized() * requiredDistance;
+        // Calculates the vector from the parents' center to his membrane towards me.
+        // This gets added to the vector calculated one call before.
+        correctedVectorToParent -= ColonyParent.Membrane
+            .GetExternalOrganelle(vectorFromParentRotated.x, vectorFromParentRotated.z)
+            .Rotated(Vector3.Up, globalParentRotation.y);
```

Comment:
The comment here makes me think that there is some underlying intermediate variable that could be made explicit for readability. Also, the resulting vector seems to be the translation to apply so that contact by external organelles is correct? I don't think it points toward parent anyways, so unless I'm mistaken this should go into a dedicated variable.

Also, you say "added" but use operator `-=`. Looking at the code it seems you add the opposite of the vector from parent's center to the membrane? 

Given that we work with vectors, I for one prefer to read vector1+(-vector2) when dealing with opposites, especially when vector2's origin is closer than the center. I find it easier to visualize, but that may not be a shared opinion.